public/app/features/alerting/unified/components/receivers/TemplateForm.tsx

```diff
@@ -92,86 +133,116 @@ export const TemplateForm = ({ existing, alertManagerSourceName, config, provena
+                {view === 'content' ? (
+                  <div>
+                    <Field error={errors?.content?.message} invalid={!!errors.content?.message} required>
+                      <div className={styles.editWrapper}>
+                        <AutoSizer>
+                          {({ width, height }) => (
+                            <TemplateEditor
+                              value={getValues('content')}
+                              width={width}
+                              height={height}
+                              onBlur={(value) => setValue('content', value)}
+                            />
+                          )}
+                        </AutoSizer>
+                      </div>
+                    </Field>
+                    <div className={styles.buttons}>
+                      {loading && (
+                        <Button disabled={true} icon="fa fa-spinner" variant="primary">
+                          Saving...
+                        </Button>
+                      )}
+                      {!loading && (
+                        <Button type="submit" variant="primary">
+                          Save template
+                        </Button>
+                      )}
+                      <LinkButton
+                        disabled={loading}
+                        href={makeAMLink('alerting/notifications', alertManagerSourceName)}
+                        variant="secondary"
+                        type="button"
+                        fill="outline"
+                      >
+                        Cancel
+                      </LinkButton>
+                    </div>
+                  </div>
+                ) : (
+                  <TemplatePreview payload={payload} templateName={watch('name')} />
+                )}
               </div>
-            </Field>
-            <div className={styles.buttons}>
-              {loading && (
-                <Button disabled={true} icon="fa fa-spinner" variant="primary">
-                  Saving...
-                </Button>
-              )}
-              {!loading && (
-                <Button type="submit" variant="primary">
-                  Save template
-                </Button>
-              )}
-              <LinkButton
-                disabled={loading}
-                href={makeAMLink('alerting/notifications', alertManagerSourceName)}
-                variant="secondary"
-                type="button"
-                fill="outline"
-              >
-                Cancel
-              </LinkButton>
             </div>
-          </div>
-          <TemplateDataDocs />
-        </div>
-      </FieldSet>
-    </form>
+            {alertManagerSourceName === GRAFANA_RULES_SOURCE_NAME && (
+              <PayloadEditor payload={payload} setPayload={setPayload} defaultPayload={DEFAULT_PAYLOAD} />
+            )}
+          </Stack>
+        </FieldSet>
+        <>
+          <ExpandableSection
```

Comment:
Maybe we can use `CollapsableSection` here?

https://developers.grafana.com/ui/latest/index.html?path=/story/layout-collapsablesection--basic